### DISCUSSION TITLE
DB and App Separate or together?

### DISCUSSION THREAD

**Discussion by ibnsina77** [dokku/dokku#4605] [Q&A] [answered]
If have a server with enough storage. Why do we have to rent some external storage? 

Could I simply not have one container for DB, another container for the App on the same server without additional mounted storage?

**Comment by Cellane** [comment, marked as answer]
This question very much confuses me, especially in tandem with your previous question #4604 😅
By default, the storage plugins don’t require you to rent any sort of external storage. Where did you get that impression from?
The storage from your app(’s container) and your database(’s container) are separate (that’s a good thing), but they’re both local.

**Reply by ibnsina77** [reply]
Okay. Thank you @Cellane. But, how can I on redeploy of the App, not lose any Database Data?
According to your answer, they are separate i.e. the Database Container should be there, but in my case it is not.

**Reply by Cellane** [reply]
@tarikmiljkovic If you use Dokku’s datastore plugins, such as [dokku/dokku-postgres](https://github.com/dokku/dokku-postgres) or [dokku/dokku-mysql](https://github.com/dokku/dokku-mysql), then the datastore will run in a separate container (as it should) with its data stored in a separate persistent volume.
